Moggles.UnitTests/FeatureTogglesTests/FeatureTogglesTests.cs

```diff
@@ -414,5 +414,126 @@ public void CreateEnvironment_A_NewEnvironmentIsBeingStoredInDB()
             _context.DeployEnvironments.FirstOrDefault(x => x.DefaultToggleValue == false).Should().NotBeNull();
         }
+
+        [TestMethod]
+        public void EditEnvironment_EnvironmentIsBeingModified()
+        {
+            //arrange
+            var app = new Application { Id = 1, AppName = "TestApp" };
+            var environment = new DeployEnvironment
+                { Application = app, ApplicationId = app.Id, EnvName = "DEV" };
+
+            _context.DeployEnvironments.Add(environment);
+            _context.SaveChanges();
+
+            var controller = new FeatureTogglesController(_context);
+
+            var updatedEnvironmentName = "QA";
+
+            var updatedEnvironment = new UpdateEnvironmentModel
+            {
+                ApplicationId = environment.ApplicationId,
+                InitialEnvName = environment.EnvName,
+                NewEnvName = updatedEnvironmentName
+            };
+
+            //act
+            var result = controller.UpdateEnvironment(updatedEnvironment);
+
+            //assert
+            result.Should().BeOfType<OkResult>();
+            _context.DeployEnvironments.FirstOrDefault(e => e.EnvName == updatedEnvironmentName).Should().NotBeNull();
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(InvalidOperationException), "Environment does not exist!")]
+        public void EditEnvironment_EnvironmentIsModifiedWithInvalidID_ThrowsInvalidOperationException()
+        {
+            //arrange
+            var app = new Application { Id = 1, AppName = "TestApp" };
+            var environment = new DeployEnvironment
+                { Application = app, ApplicationId = app.Id, EnvName = "DEV" };
+
+            _context.DeployEnvironments.Add(environment);
+            _context.SaveChanges();
+           
+            var controller = new FeatureTogglesController(_context);
+            
+            var updatedEnvironmentName = "QA";
+
+            var updatedEnvironment = new UpdateEnvironmentModel
+            {
+                ApplicationId = environment.ApplicationId + 1,
+                InitialEnvName = environment.EnvName,
+                NewEnvName = updatedEnvironmentName
+            };
+
+            //act
+            controller.UpdateEnvironment(updatedEnvironment);
+
+            //assert
+            //throws InvalidOperationException
+        }
+
+        [TestMethod]
+        public void DeleteEnvironment_EnvironmentIsDeleted_FeatureToggleStatusesAreDeleted()
+        {
+            //arrange
+            var app = new Application { Id = 1, AppName = "TestApp" };
+            var environment = new DeployEnvironment
+                { Application = app, ApplicationId = app.Id, EnvName = "TestEnv" };
+
+            var controller = new FeatureTogglesController(_context);
+
+            var firstFeatureStatus = new FeatureToggleStatus { Enabled = false, Id = 1, IsDeployed = false, Environment = environment, EnvironmentId = environment.Id };
```

Comment:
What about the feature toggles? Don't you need them in order to create a feature toggle status?
I think you should also add a test that check that the feature toggles are not removed together with the statuses